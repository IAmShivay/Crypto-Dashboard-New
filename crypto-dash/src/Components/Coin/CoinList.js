import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Coin from './Coin';
import { useSelector } from 'react-redux';

function CoinList() {

    const [coins, setCoins] = useState([]);

    const currencyVal = useSelector((state) => state.CryptoCurrency);

    const searchVal = useSelector((state) => state.search);


    useEffect(() => {
        axios
            .get(
                `https://api.coingecko.com/api/v3/coins/markets?vs_currency=${currencyVal}&order=market_cap_desc&per_page=50&page=1&sparkline=true`
            )
            .then((res) => {
                setCoins(res.data);
                console.log('COIN LIST', res.data);
            })
            .catch((error) => console.log(error));
    }, [currencyVal]);

    const filteredCoins = coins.filter((coin) =>
        coin.name.toLowerCase().includes(searchVal.toLowerCase())
    );

    return (
        <div className="py-5 ">
            <div className="px-4 mb-8 xl:mb-9 lg:mb-6 font-bold text-xl tracking-wider text-stone-600">
                <h1 className='dark:text-gray-50'>Cryptocurrency by market cap</h1>
            </div>

            <div className="h-[44rem] lg:h-[41rem] xl:h-[44rem] overflow-y-scroll scrollbar-hide">
                {filteredCoins.map((coin) => {
                    return (
                        <Coin
                            key={coin.id}
                            name={coin.name}
                            marketcap={coin.market_cap}
                            image={coin.image}
                            priceChange={coin.price_change_percentage_24h}
                            currencyVal={currencyVal}
                        />
                    );
                })}
            </div>
        </div>
    );
}

export default CoinList;
