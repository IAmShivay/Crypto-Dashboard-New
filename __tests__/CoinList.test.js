import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CoinList from '../src/Components/Coin/CoinList';

const mockStore = configureStore();

test('CoinList renders correctly', () => {
    const store = mockStore({
        CryptoCurrency: 'usd',
        search: '',
    });

    const { asFragment } = render(
        <Provider store={store}>
            <CoinList />
        </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
});
