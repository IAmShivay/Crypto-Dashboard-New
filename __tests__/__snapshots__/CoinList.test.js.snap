// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="py-5 "
  >
    <div
      class="px-4 mb-8 xl:mb-9 lg:mb-6 font-bold text-xl tracking-wider text-stone-600"
    >
      <h1
        class="dark:text-gray-50"
      >
        Cryptocurrency by market cap
      </h1>
    </div>
    <div
      class="h-[44rem] lg:h-[41rem] xl:h-[44rem] overflow-y-scroll scrollbar-hide"
    />
  </div>
</DocumentFragment>
`;
